sway_libs/src/fixed_point/ufp/ufp128/ufp128.sw

```diff
@@ -0,0 +1,234 @@
+library ufp128;
+//! A wrapper around U128 type for a library for Sway for mathematical functions operating with signed 64.64-bit fixed point numbers. 
+use std::u128::U128;
+use std::u256::U256;
+use std::assert::assert;
+use std::math::{Exponent, Exponentiate, Root};
+use std::revert::revert;
+
+pub struct UFP128 {
+    value: U128,
+}
+
+impl From<(u64, u64)> for UFP128 {
+    fn from(int_fract_tuple: (u64, u64)) -> Self {
+        Self {
+            value: U128::from(int_fract_tuple),
+        }
+    }
+
+    fn into(self) -> (u64, u64) {
+        self.value.into()
+    }
+}
+
+impl UFP128 {
+    pub fn zero() -> Self {
+        Self {
+            value: U128::from((0, 0)),
+        }
+    }
+
+    /// The smallest value that can be represented by this type.
+    pub fn min() -> Self {
+        Self {
+            value: U128::min(),
+        }
+    }
+
+    /// The largest value that can be represented by this type,
+    pub fn max() -> Self {
+        Self {
+            value: U128::max(),
+        }
+    }
+
+    /// The size of this type in bits.
+    pub fn bits() -> u32 {
+        128
+    }
+}
+
+impl UFP128 {
+    /// Creates UFP64 that correponds to a unsigned integer
+    pub fn from_uint(uint: u64) -> Self {
+        Self {
+            value: U128::from((uint, 0)),
+        }
+    }
+}
+
+impl core::ops::Eq for UFP128 {
+    fn eq(self, other: Self) -> bool {
+        self.value == other.value
+    }
+}
+
+impl core::ops::Ord for UFP128 {
+    fn gt(self, other: Self) -> bool {
+        self.value > other.value
+    }
+
+    fn lt(self, other: Self) -> bool {
+        self.value < other.value
+    }
+}
+
+impl core::ops::Add for UFP128 {
+    /// Add a UFP128 to a UFP128. Panics on overflow.
+    fn add(self, other: Self) -> Self {
+        UFP128 {
+            value: self.value + other.value,
+        }
+    }
+}
+
+impl core::ops::Subtract for UFP128 {
+    /// Subtract a UFP128 from a UFP128. Panics of overflow.
+    fn subtract(self, other: Self) -> Self {
+        // If trying to subtract a larger number, panic.
+        assert(!(self.value < other.value));
```

Comment:
```suggestion
        assert((self.value >= other.value));
```

Reply:
No method named "ge" found for type "U128", so it won't work. I'll add it to sway, but as this is a minor issue, I'd like this part to be merged as is for now.